OSDOCS-10601: Updates to Custom admin group access for netobserv (#78162)
openshift/openshift-docs: modules/logging-loki-log-access.adoc

// tag::CustomAdmin[]
== Custom admin group access

// tag::LokiMode[]

> If you have a large deployment with several users who require broader permissions, you can create a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR are considered administrators. 

Reviewer: ```suggestion
If you have a large deployment with several users who require broader permissions, you can create a custom group to configure with `adminGroup` field of LokiStack. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR are considered administrators. 
```
I don't adminGroup field doesn't create group itself, a group has to be created prior before specifying in adminGroup field.

Author: @memodi This paragraph is for Loki docs. The following paragraph with the `/ tag::NetObservMode[]` is for NetObserv, and the piece you commented about is slightly different: 

> If you need to see cluster-wide logs without necessarily being an administrator, or if you already have any group defined that you want to use here, you can specify a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as administrators.

Reviewer: I see, understood, then ignore my comment.

> If you have a large deployment with several users who require broader permissions, you can create a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR are considered administrators. 

Reviewer 2: Since this is the first mention of CR in the section:
```suggestion
If you have a large deployment with several users who require broader permissions, you can create a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` custom resource (CR) are considered administrators. 
```

Author: Addressed in comment below.

// end::LokiMode[]

// tag::NetObservMode[]

> If you need to see cluster-wide logs without being an administrator, you can create a custom group using the `adminGroup` field.  Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR are considered admins. 

Reviewer 3: ```suggestion
If you need to see cluster-wide logs without being an administrator, you can create a custom group using the `adminGroup` field.  Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as admins. 
```

wdyt? It's to not make people think this would turn these users into admins

Author: I agree this is a good point to clarify, thanks for bringing it up 👍

> If you need to see cluster-wide logs without being an administrator, you can create a custom group using the `adminGroup` field.  Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR are considered admins. 

Reviewer 3: Actually, there's another use-case that is relevant for this config, and I've seen customers mentioning it, it's when they already have some admin groups defined, but not named "cluster-admin", and they'd like to re-use it rather than creating a new one, so they can also do the same here.

So we could formulate it like:

```suggestion
If you need to see cluster-wide logs without necessarily being an administrator, or if you already have a group defined for cluster admins that you would like to use here, you can specify a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as admins. 
```

Author: Awesome, thanks for adding these details and the extra use case.

> If you need to see cluster-wide logs without necessarily being an administrator, or if you already have any group defined that you want to use here, you can specify a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as administrators. 

Reviewer 2: Not fully sure how tags work, but if this is the first instance of CR in whichever section this is included in, then same comment as above:
```suggestion
If you need to see cluster-wide logs without necessarily being an administrator, or if you already have any group defined that you want to use here, you can specify a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` custom resource (CR) have the same read access to logs as administrators. 
```

Author: Yeah I was gonna say, the other one you commented on is for the Loki docs, so I think this one is actually the one I'll need to change, since its the content I own.

> If you need to see cluster-wide logs without necessarily being an administrator, or if you already have any group defined that you want to use here, you can specify a custom group using the `adminGroup` field. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as administrators. 

Reviewer 2: I think I'm personally getting a little tripped up by the phrase "or if you already have any group defined that you want to use here", potentially because the phrase is mentioned before the context of what "use here" means (I'm guessing the action of specifying a custom group using the `adminGroup` field).

Would it be possible to remove that phrase from this sentence and make it its own sentence somewhere later in the paragraph?

For example:
```suggestion
If you need to see cluster-wide logs without necessarily being an administrator, you can specify a custom group using the `adminGroup` field. You can also specify any group that is already defined. Users who are members of any group specified in the `adminGroups` field of the `LokiStack` CR have the same read access to logs as administrators. 
```

Not sure if my suggestion is still technically accurate, but the basis of my suggestion is seeing if the phrase can be moved somewhere after the first sentence.

Author: I think your suggestion hangs here, so I'll take it. I agree, the interrupting phrase definitely makes this sentence less clear.

// end::NetObservMode[]

// tag::LokiMode[] 

> Administrator users have access to all application logs in all namespaces, if they also get assigned the `cluster-logging-application-view` role.

Reviewer 2: Suggestion - just sounds a little more formal, up to you:
```suggestion
Administrator users have access to all application logs in all namespaces, if they are also assigned the `cluster-logging-application-view` role.
```

Author: Since this is only in the Loki section, I'm not going to touch this, but I agree.
